File: core/recc/packet/cvt/group.py
# -*- coding: utf-8 -*-

from recc.packet.group import GroupA
from recc.database.struct.group import Group


def group_to_answer(group: Group) -> GroupA:
    group_slug = group.slug if group.slug else ""
    return GroupA(
        slug=group_slug,
        name=group.name,
        description=group.description,
        features=group.features,
        visibility=group.visibility,
        extra=group.extra,
        created_at=group.created_at,
        updated_at=group.updated_at,
    )

File: api/src/packet/group.ts
export interface GroupA {
  slug: string;
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  created_at?: string;
  updated_at?: string;
}

export interface CreateGroupQ {
  slug: string;
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

export interface UpdateGroupQ {
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}
